File: bookworm-backend/spring-boot-ecommerce/src/main/java/com/bookworm/ecommerce/dto/Purchase.java
package com.bookworm.ecommerce.dto;


import com.bookworm.ecommerce.entity.Address;
import com.bookworm.ecommerce.entity.Customer;
import com.bookworm.ecommerce.entity.Order;
import com.bookworm.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address billingAddress;
    private Address shippingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: bookworm_front/src/app/core/models/purchase.ts
import { Customer } from './customer';
import { Address } from './address';
import { OrderItem } from './order-item';
import { Order } from './order';

export class Purchase {
  customer: Customer;
  shippingAddress: Address;
  billingAddress: Address;
  order: Order;
  orderItems: OrderItem[];
}
